executor/aggregate.go

```diff
@@ -736,8 +743,9 @@ func (e *HashAggExec) waitPartialWorkerAndCloseOutputChs(waitGroup *sync.WaitGro
 	}
 }
 
-func (e *HashAggExec) waitFinalWorkerAndCloseFinalOutput(waitGroup *sync.WaitGroup) {
+func (e *HashAggExec) waitAllWorkerAndCloseFinalOutput(waitGroup *sync.WaitGroup, waitGroup2 *sync.WaitGroup) {
 	waitGroup.Wait()
+	waitGroup2.Wait()
```

Comment:
Why change this?

Reply:
revert.

Reply:
When there causes any error, HashAggExec will close finishCh, and all goroutine will try to exit.  We can't guarantee fetchChildWorker and partialWorker will exit before finalWorker.
If all finalWorkers exit first and close finalOutputCh, recoveryHashAgg() in fetchChildWorker/partialWorker will try to send message to a closed chan if there are some panic..

In my random panic test, I find the data race.